Bastien Durand/week_01/calculator/calculator.js

```diff
@@ -0,0 +1,71 @@
+// # The Calculator
+
+// ## Part 1
+//- Write a function called squareNumber that will take one argument (a number),
+//  square that number, and return the result.
+//  It should also log a string like "The result of squaring the number 3 is 9."
+
+//- Write a function called halfNumber that will take one argument (a number),
+//  divide it by 2, and return the result. It should also log a string like "Half of 5 is 2.5.".
+
+//- Write a function called percentOf that will take two numbers,
+// figure out what percent the first number represents of the second number, and return the result.
+// It should also log a string like "2 is 50% of 4."
+
+//- Write a function called areaOfCircle that will take one argument (the radius),
+//  calculate the area based on that, and return the result.
+//  It should also log a string like "The area for a circle with radius 2 is 12.566370614359172."
+
+//-Bonus: Round the result so there are only two digits after the decimal.
+
+//PART 1 ANSWERS
+
+
+const squareNumber = function (num) {
+  squaredResult = num * num
+  console.log(`The result of squaring the number ${num} is ${squaredResult}`)
+};
+
+const halfNumber = function (num) {
+  halfResult = (num / 2);
+  console.log(`Half of ${num} is ${halfResult}`)
+};
+
+const percentOf = function (num1, num2) {
+  percentResult = (Math.floor((num1 / num2) * 100));
+  console.log(`${num1} is ${percentResult}% of ${num2}`)
+};
+
+const areaOfCircle = function (radius) {
+  areaResult = Math.PI * radius * radius
+  console.log(`The area for a circle with radius ${radius} is ${areaResult}`)
+};
+
+
+squareNumber(9);
+halfNumber(5);
+percentOf(2, 4);
+areaOfCircle(2);
+
+
+// // ## Part 2
+// // Write a function that will take one argument (a number) and perform the following operations,
+//  using the functions you wrote earlier1:
+// // - Take half of the number and store the result.
+// // - Square the result of #1 and store that result.
+// // - Calculate the area of a circle with the result of #2 as the radius.
+// // - Calculate what percentage that area is of the squared result (#3).
+
+//
+
+//PART 2 ANSWERS
+const calculateAll = function (num) {
+  result = halfNumber(num);
+  num.result = squareNumber(num);
+  let squared = this.result
+  num.result = areaOfCircle(num);
+  num.result = percentOf(num, squared);
+
+};
+//DIDN'T WORK RAN OUT OF TIME
```

Comment:
The reason this isn't working is due to your use of the functions you've written earlier. Because all of your earlier functions (e.g. halfNumber) don't have a return statement (they simply use console.log()) they don't actually give you a value back. 
Keeping in mind what we went through today in class about return statements, give this another go.